profile.py

```diff
@@ -150,8 +172,10 @@ def right_penumbra(self, upper=0.9, lower=0.1):
     def center(self, level=0.5):
         return 0.5 * (self.x_at_y(level) + self.x_at_y(level, reverse=True))
 
-    def mirror(self):
-        self.y = self.y[::-1]
+    def mirror(self, m=0):
+        # self.y = 2*m - self.y
+        self.x = 2*m - self.x
+        self.fix_profile()
```

Comment:
well, if given Y is not inside the domain of profile inverting the array could be used (and probably would be much more efficient then sort). On the other hand when given Y is somewhere in the domain of profile (X) it would require some extra operations, although I think it is possible and if I figure it out I will change it to use only inverting function (a[::-1])